src/Configuration.jl

```diff
@@ -66,8 +66,8 @@ The HTTP server options. See [`SecurityOptions`](@ref) for additional settings.
 # Keyword arguments
 
 - `host::String = "$HOST_DEFAULT"` Set to `"127.0.0.1"` (default) to run on *localhost*, which makes the server available to your computer and the local network (LAN). Set to `"0.0.0.0"` to make the server available to the entire network (internet).
-- `port::Union{Nothing,Integer} = $PORT_DEFAULT` When specified, this port will be used for the server.
-- `port_hint::Integer = $PORT_HINT_DEFAULT` If the other setting `port` is not specified, then this setting (`port_hint`) will be used as the starting point in finding an available port to run the server on. 
+- `port::Union{Nothing,Int} = $PORT_DEFAULT` When specified, this port will be used for the server.
```

Comment:
Unfortunately this would be a breaking change (this is public API). I would say to leave it at `Integer` unless we have good evidence that it would improve performance.

Reply:
I see. I will drop this commit, since the benefit is marginal, mostly on precompilation time.

As an aside, I think there could be some better inference and shorter precompilation time if we remove more Union splitting. But since this is breaking I think it would be good to attempt it all at once, and see what kind of improvements are possible and if such a change is worth is.